packages/ecc-utils-design/src/components/details/details.ts

```diff
@@ -193,100 +145,190 @@ export default class EccUtilsDesignDetails extends LitElement {
+    data: any,
+    copyLabel: string,
+    successLabel: string,
+    errorLabel: string
+  ) {
+    this.requestUpdate();
+    return html` <span class="label-copy">
+      <span>${label}</span>
+      ${copy
+        ? html`
+            <sl-copy-button
+              value=${data}
+              copy-label=${copyLabel}
+              success-label=${successLabel}
+              error-label=${errorLabel}
+            ></sl-copy-button>
+          `
+        : html``}
+    </span>`;
   }
 
-  private _renderObjectItem(
-    [key, value]: [string, string],
-    child: Children
+  private _renderData(
+    data: string,
+    label: string,
+    copy = false
   ): TemplateResult {
+    if (data === null || data === undefined) return html``;
+    this.requestUpdate();
     return html`
-      <div class="object-item">
-        <div class="object-key">${key}:</div>
-        <div id="${child.key}" class="object-value">${value}</div>
-        ${child.textOptions?.copy &&
-        html`<sl-copy-button
-          class="copy-button"
-          from=${child.key}
-        ></sl-copy-button>`}
-      </div>
-    `;
-  }
-
-  private _renderObject(child: Children): TemplateResult {
-    return html`
-      <div class="details-container details-container-object">
-        <div class="label">${child.label}</div>
-        <div class="object-value-container">
-          ${Object.entries(child.value).map((entry) =>
-            this._renderObjectItem(entry, child)
+      <div part="data-container" class="container data-container">
+        <div part="label" class="label">
+          ${this._renderLabel(
+            label,
+            copy,
+            data,
+            `Copy`,
+            `${label} copied!`,
+            "Error"
           )}
         </div>
+        <div part="value" class="value">${data}</div>
       </div>
     `;
   }
 
-  private _renderArrayValue(value: string): TemplateResult {
-    return html`<div class="value">${value}</div>`;
-  }
-
-  private _renderArrayPill(value: string): TemplateResult {
-    return html`<sl-tag size="medium" pill>${value}</sl-tag>`;
+  private _renderArray(
+    data: Array<any>,
+    label: string,
+    copy = false
+  ): TemplateResult {
+    if (data === null || data === undefined || data.length === 0) return html``;
+    return html`
+			<div part="container" class="container">
+				<sl-details>
+					<div part="summary-container" slot="summary" class="summary">
+          ${this._renderLabel(
+            label,
+            copy,
+            data,
+            `Copy JSON`,
+            `${label} copied!`,
+            "Error"
+          )}
+						</span>
+					</div>
+          <div class="panel-container">
+            ${data.map((value, index) => {
+              const newLabel = `${label} ${index + 1}`;
+              if (value === null || value === undefined) {
+                return null; // Skip rendering for null or undefined values
+              }
+              if (Array.isArray(value)) {
```

Comment:
you're using very similar logic in 2 other places, the `_renderObject` method and the `_renderField` method. Perhaps you could convert it to a function. Also, is there no option to copy sub-objects or sub-arrays?